### DISCUSSION TITLE
MC TemporalDenoise add it back

### DISCUSSION THREAD

**Discussion by joh986** [staxrip/staxrip#1392] [General]
HI, new b here , just wondering if there anyway to add back MC Temporal Denoise to the apps , so it will work ? instead of getting a error  code cant load ? Any help would b great

 
![Screenshot 2024-06-12 003929](https://github.com/staxrip/staxrip/assets/172405432/6de446c7-29ff-4235-914f-f147e53553d8)

**Comment by Dendraspis** [comment]
You have to import those two filters before the call of `MCTemporalDenoise()`:
`.\Apps\Plugins\AVS\FFT3DFilter\fft3dfilter.dll`
`.\Apps\Plugins\Dual\f3kdb Neo\neo-f3kdb.dll`

This will be fixed with the next release...

**Reply by joh986** [reply]
ok thank u 4 your reply ,

**Reply by Dendraspis** [reply]
Can you check if this one works for you without adding imports?

[StaxRip v2.40.1 (EXE ONLY)](https://ufile.io/ew5qqvcb)

**Reply by joh986** [reply]
move them both over into file with no luck 
![Screenshot tttt](https://github.com/staxrip/staxrip/assets/172405432/5a696c73-0a8f-45a5-b1a0-c28944c19a12)

**Reply by Dendraspis** [reply]
Does your system have a (dedicated) GPU or is it a server?

**Reply by joh986** [reply]
GPU

**Reply by joh986** [reply]
> GPU

l just might wait 2 u put up a new release one with the file in it , as l said am a new b at this , and thanks again 4 your help , regrads john

**Reply by Dendraspis** [reply]
> l just might wait 2 u put up a new release one with the file in it

The file is already there.

Can you right-click on the filters list and then click on `Preview Code...` and post the code here?

**Comment by joh986** [comment]


**Reply by Dendraspis** [reply]
The code above would also have been interesting.  😅 

You can also copy & paste it.  😉

**Comment by joh986** [comment]
1: AddAutoloadDir("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\FrameServer\AviSynth\plugins\")
  2: LoadPlugin("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\Deblock\Deblock.dll")
  3: Import("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\Deblock_QED\Deblock_QED.avsi")
  4: LoadPlugin("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\EEDI2\EEDI2.dll")
  5: LoadPlugin("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\FFT3DGPU\FFT3dGPU.dll")
  6: LoadPlugin("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\GradFun2DB\gradfun2db.dll")
  7: Import("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\GradFun2DB\GradFun2DBmod.avsi")
  8: Import("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\LSFmod\LSFmod.avsi")
  9: LoadPlugin("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\masktools2\masktools2.dll")
 10: LoadPlugin("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\mvtools2\mvtools2.dll")
 11: LoadPlugin("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\RgTools\RgTools.dll")
 12: LoadPlugin("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\FFT3DFilter\fft3dfilter.dll")
 13: LoadPlugin("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\Dual\FFT3D Neo\neo-fft3d.dll")
 14: LoadPlugin("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\TMM2\TMM2.dll")
 15: Import("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\DAA3Mod\daa3mod.avsi")
 16: LoadPlugin("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\DFTTest\dfttest.dll")
 17: LoadPlugin("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\Dual\DFTTest Neo\neo-dfttest.dll")
 18: Import("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\Dither\dither.avsi")
 19: LoadPlugin("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\Dither\dither.dll")
 20: LoadPlugin("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\EEDI3\eedi3.dll")
 21: LoadPlugin("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\HQDN3D\Hqdn3d.dll")
 22: LoadPlugin("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\JPSDR\Plugins_JPSDR.dll")
 23: LoadPlugin("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\Dual\KNLMeansCL\KNLMeansCL.dll")
 24: LoadPlugin("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\MedianBlur2\MedianBlur2.dll")
 25: LoadPlugin("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\Dual\MiniDeen\neo-minideen.dll")
 26: LoadPlugin("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\MSharpen\msharpen.dll")
 27: LoadPlugin("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\RT_Stats\RT_Stats_x64.dll")
 28: LoadPlugin("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\SangNom2\SangNom2.dll")
 29: LoadPlugin("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\TIVTC\TIVTC.dll")
 30: LoadPlugin("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\TTempSmooth\TTempSmooth.dll")
 31: LoadPlugin("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\vsMSharpen\vsMSharpen.dll")
 32: LoadPlugin("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\vsTTempSmooth\vsTTempSmooth.dll")
 33: Import("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\MCTemporalDenoise\MCTemporalDenoise.avsi")
 34: Import("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\Zs_RF_Shared\Zs_RF_Shared.avsi")
 35: # can be configured at: Tools > Settings > Source Filters
 36: MCTemporalDenoise(settings="medium")

**Reply by Dendraspis** [reply]
Line 5 clearly shows:
`LoadPlugin("C:\Users\johnh\Desktop\Set up Files\StaxRip-v2.40.0-x64\Apps\Plugins\AVS\FFT3DGPU\FFT3dGPU.dll")`

So I guess the GPU, that is used by the plugin, doesn't fit it's requirements.  🤔 
Which GPU is it? And do you work on that system locally or remotely, for example via RDP?

**Comment by joh986** [comment]
Thats weird had a early version of staxrip and that work fine , but since the updates came MC TemporalDenoise dosent work , Did they disable  it ?, And should l go back to a earlier version of staxrip to get  MC TemporalDenoise back ? Which GPU is it?and example via RDP ?

**Reply by Dendraspis** [reply]
> Did they disable it ?

Still don't know what you mean by that.

> And should l go back to a earlier version of staxrip to get MC TemporalDenoise back

Please tell my which older version does work while `v2.40` does not.

> Which GPU is it?

Which GPU is built in that system?

> and example via RDP ?

Do you work via RDP or similar on that system or are you sitting in front of it?

**Comment by joh986** [comment]
GPU 0

	Intel(R) UHD Graphics

	Driver version:	27.20.100.9079
	Driver date:	11/12/2020
	DirectX version:	12 (FL 12.1)
	Physical location:	PCI bus 0, device 2, function 0

	Utilization	27%
	Dedicated GPU memory	
	Shared GPU memory	0.6/5.9 GB
	GPU Memory	0.6/5.9 GB
![image](https://github.com/staxrip/staxrip/assets/172405432/8ca1a96a-f36f-4769-960d-9096d3ccf82e)

Did they disable it ? Yes The did

**Reply by Dendraspis** [reply]
The only requirement is a `Directx 9 graphics card` and none of those filters and plugins has changed. Maybe you should make sure that `DirectX` is installed.  🤔 

For me everything works:

![image](https://github.com/staxrip/staxrip/assets/50659978/26f4afa8-1be3-4416-957e-b13171f26291)

**Comment by joh986** [comment]
Dendraspis  like to say , thank u for your help very much appreciated , works good now with this  directx_Jun2010_redist  cant thank u enough pointing me into the right direction , thanks again john
